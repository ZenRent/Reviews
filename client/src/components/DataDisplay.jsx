import React from 'react';
import styled from 'styled-components';

const Wrapper = styled.div`
display: flex;
flex-flow: row nowrap;
margin-left: 28%;
margin-right: 28%;
color: #222222;
`;

const Heading = styled.h1`
flex: 1;
${'' /* border: 1px dotted; */}
text-align: left;
padding: 5px;
font-weight: bold;
font-size:16px;
`;

const Cell = styled.div`
flex: 1;
${'' /* border: 1px dotted; */}
text-align: left;
padding: 5px;
font-size: 13px;
`;

const ReviewStar = styled.img`
width:20px;
height: auto;
margin-right: 7px;
`;

const RatingBar = styled.img`
width:150px;
height: 15px;
margin-right: 7px;
`;

const DataDisplay = (props) => {
  return (<div>
    <Wrapper><Heading><ReviewStar src='red-star.png' />4.94 (161 reviews)</Heading></Wrapper>
    <Wrapper><Cell>Cleanliness</Cell><Cell><RatingBar src='RatingBar.png' />5.0</Cell>
      <Cell>Accuracy</Cell><Cell><RatingBar src='RatingBar.png' />4.5</Cell></Wrapper>
    <Wrapper><Cell>Communication</Cell><Cell><RatingBar src='RatingBar.png' />4.8</Cell><Cell>Location</Cell><Cell><RatingBar src='RatingBar.png' />4.9</Cell></Wrapper>
    <Wrapper><Cell>Check-in</Cell><Cell><RatingBar src='RatingBar.png' />4.6</Cell><Cell>Value</Cell><Cell><RatingBar src='RatingBar.png' />5.0</Cell></Wrapper>
  </div>);
};

export default DataDisplay;
