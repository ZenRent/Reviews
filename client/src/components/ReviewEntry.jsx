import React from 'react';
import styled from 'styled-components';

const ReviewContainer = styled.div`
display: flex;
margin: 20px 20px 3px 20px;
margin-left: 28%;
margin-right: 28%;
padding: 4px;
font-size: 13px;
color: #222222;
grid-gap: 5px;
width: 470px;
letter-spacing: .75px;
width: 50%;
`;

const VerticalContainer = styled.div`
display: flex;
flex-direction: column;
align-items: flex-start;
`;

const Main = styled.div`
display: flex;
margin: 3px 20px 3px 20px;
margin-left: 28%;
margin-right: 28%;
padding: 4px;
font-size: 13px;
color: #222222;
grid-gap: 10px;
width: 370px;
letter-spacing: .75px;
`;

const Img = styled.img`
height: 50px;
width: 50px;
border-radius: 50%;
justify-content: flex-start;
${'' /* width: ${(props) => props.size / 12 * 100}vw; */}
`;

const Paragraph = styled.p`
text-align: left;
justify-content: flex-start;
line-height: 1.5;
`;

const User = styled.div`
font-weight: 900;
color: black;
font-size: 12.3px;

`;
const Date = styled.div`
font-weight: lighter;
font-size: 12px;
`;

const HighLight = styled.p`
color: yellow;
`;

const ReviewEntry = (props) => {
  let { Body, Username, DateTime, Image } = props.review.Reviews[0];

  return (<div><ReviewContainer>
    <Img src={Image} ></Img>
    <VerticalContainer>
      <User>{Username}</User>
      <Date>{DateTime}</Date>
    </VerticalContainer>
  </ReviewContainer>
    <Main>
      <Paragraph>{Body}</Paragraph>
    </Main></div>);
};

export default ReviewEntry;