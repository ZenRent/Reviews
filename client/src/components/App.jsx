import React from 'react';
import axios from 'axios';
import styled from 'styled-components';
import DataDisplay from './DataDisplay.jsx';
import ReviewList from './ReviewList.jsx';
//Test

const ModalBtn = styled.button`
display: flex;
margin-left: 10%;
border: 1px solid;
padding: 8px;
background-color: white;
border-radius: 7px;
`;

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      reviews: [],
    };

    this.getReviews = this.getReviews.bind(this);
  }

  componentDidMount() {
    this.getReviews();
  }

  getReviews() {
    console.log('hit getReviews');
    axios.get('/reviews')
      .then((response) => {
        console.log('response.data', response.data);
        const targetReviews = [];
        response.data.forEach((current) => {
          if (current.Listing === '69664 Daija Wall') {
            targetReviews.push(current);
          }
        });
        this.setState({ reviews: targetReviews });
      })
      .catch((error) => {
        console.log(error);
      });
  }

  render() {
    return (
      <div>
        <DataDisplay />
        <ReviewList reviews={this.state.reviews} />
        <ModalBtn>Show all Reviews</ModalBtn>
      </div>
    );
  }
}

export default App;
