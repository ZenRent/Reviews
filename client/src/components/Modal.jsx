import React from 'react';
import styled from 'styled-components';
import ReviewList from './ReviewList.jsx';
import DataDisplay from './DataDisplay.jsx';
import SearchBar from './SearchBar.jsx';

const ModalObj = styled.div`
display: flex;
flex-direction: column;
align-items: flex-start;
justify-content: flex-start;
width: 50%;
height: 90%;
border: 1px solid;
padding: 8px;
background-color: white;
border-radius: 7px;
${'' /* allow modal to accomodate all of content at present view */}
min-height: 100px;
overflow: hidden;
${'' /* max height and overflow allow scrollable */}
max-height: calc(100vh - 200px);
    overflow-y: auto;

position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
`;

const XCloseBtn = styled.button`

padding: 2;
border: none;
background: none;
cursor:pointer;
color: #4c4c4c;
font-size: 12px;
font-weight: bold;
&:hover {
border-radius: 30px;
background: #BEBEBE;
  }
`;

const StickyHeader = styled.div`

    ${'' /* top: 0;
    left: 0;
    overflow: hidden;
    width: 100%;
    height: 10%;
    position: fixed;
    background-color:red;
    border: 2px Solid; */}
`;



class Modal extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      TargetString: '',
    }
    this.filterSearch = this.filterSearch.bind(this);
  }

  filterSearch(SearchInput) {
    console.log("SearchInput", SearchInput);
    this.setState({ TargetString: SearchInput });
  }


  render() {
    return (
      <ModalObj>
        <StickyHeader />
        <XCloseBtn onClick={() => this.props.closeModal()}>X</XCloseBtn>
        <SearchBar filterSearch={this.filterSearch} />
        <StickyHeader />
        <DataDisplay />
        <ReviewList TargetString={this.state.TargetString} reviews={this.props.reviews} view={this.props.view} />
      </ModalObj>)
  }
}

export default Modal;