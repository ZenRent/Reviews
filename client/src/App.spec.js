import React from 'react';
import { configure, shallow } from "enzyme";
import renderer from 'react-test-renderer';
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });
import App from './components/App';
import Modal from './components/Modal';
import ReviewEntry from './components/ReviewList';
import DataDisplay from './components/DataDisplay';


describe('Fundamental Tests', () => {
  it('Jester is running.', () => {
    expect(true).toEqual(true);
  });

  it("renders App without crashing", () => {
    shallow(<App />);
  });

  it("renders Modal without crashing", () => {
    shallow(<Modal />);
  });
  it("renders DataDisplay without crashing", () => {
    shallow(<DataDisplay />);
  });
});

describe('Advanced, Get Snapshot', () => {
  it('Shallow rendering', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
