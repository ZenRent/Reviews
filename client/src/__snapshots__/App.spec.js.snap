// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Advanced Shallow rendering 1`] = `
<div>
  <div>
    <div
      className="sc-bdfBwQ faNODT"
    >
      <h1
        className="sc-gsTCUz jrfRmK"
      >
        4.94 (161 reviews)
      </h1>
    </div>
    <div
      className="sc-bdfBwQ faNODT"
    >
      <div
        className="sc-dlfnbm iLQnHU"
      >
        Cleanliness
      </div>
      <div
        className="sc-dlfnbm iLQnHU"
      >
        **********5.0
      </div>
      <div
        className="sc-dlfnbm iLQnHU"
      >
        Accuracy
      </div>
      <div
        className="sc-dlfnbm iLQnHU"
      >
        **********4.5
      </div>
    </div>
    <div
      className="sc-bdfBwQ faNODT"
    >
      <div
        className="sc-dlfnbm iLQnHU"
      >
        Communication
      </div>
      <div
        className="sc-dlfnbm iLQnHU"
      >
        **********4.8
      </div>
      <div
        className="sc-dlfnbm iLQnHU"
      >
        Location
      </div>
      <div
        className="sc-dlfnbm iLQnHU"
      >
        **********4.9
      </div>
    </div>
    <div
      className="sc-bdfBwQ faNODT"
    >
      <div
        className="sc-dlfnbm iLQnHU"
      >
        Check-in
      </div>
      <div
        className="sc-dlfnbm iLQnHU"
      >
        **********4.6
      </div>
      <div
        className="sc-dlfnbm iLQnHU"
      >
        Value
      </div>
      <div
        className="sc-dlfnbm iLQnHU"
      >
        **********5.0
      </div>
    </div>
  </div>
  <ul
    className="sc-iBPRYJ hRnkiD"
  />
  <button
    className="sc-pFZIQ iNoUAp"
    onClick={[Function]}
  >
    Show all Reviews
  </button>
</div>
`;

exports[`Advanced, Get Snapshot Shallow rendering 1`] = `
<div>
  <div>
    <div
      className="sc-bdfBwQ faNODT"
    >
      <h1
        className="sc-gsTCUz jrfRmK"
      >
        4.94 (161 reviews)
      </h1>
    </div>
    <div
      className="sc-bdfBwQ faNODT"
    >
      <div
        className="sc-dlfnbm iLQnHU"
      >
        Cleanliness
      </div>
      <div
        className="sc-dlfnbm iLQnHU"
      >
        **********5.0
      </div>
      <div
        className="sc-dlfnbm iLQnHU"
      >
        Accuracy
      </div>
      <div
        className="sc-dlfnbm iLQnHU"
      >
        **********4.5
      </div>
    </div>
    <div
      className="sc-bdfBwQ faNODT"
    >
      <div
        className="sc-dlfnbm iLQnHU"
      >
        Communication
      </div>
      <div
        className="sc-dlfnbm iLQnHU"
      >
        **********4.8
      </div>
      <div
        className="sc-dlfnbm iLQnHU"
      >
        Location
      </div>
      <div
        className="sc-dlfnbm iLQnHU"
      >
        **********4.9
      </div>
    </div>
    <div
      className="sc-bdfBwQ faNODT"
    >
      <div
        className="sc-dlfnbm iLQnHU"
      >
        Check-in
      </div>
      <div
        className="sc-dlfnbm iLQnHU"
      >
        **********4.6
      </div>
      <div
        className="sc-dlfnbm iLQnHU"
      >
        Value
      </div>
      <div
        className="sc-dlfnbm iLQnHU"
      >
        **********5.0
      </div>
    </div>
  </div>
  <ul
    className="sc-iBPRYJ hRnkiD"
  />
  <button
    className="sc-pFZIQ iNoUAp"
    onClick={[Function]}
  >
    Show all Reviews
  </button>
</div>
`;
